migration from r754 to r1295

guitoulefoux opened the pull request:
I encountered a lot of problem to go from r754 to r1295.
Here is what I did to solve my problems.

guitoulefoux committed:
mysql invalid default value for 'expiry_date'

For a migration from r754 to r1295 (MySQL version 5.7.35) I have the error.

The following modification corrects my problem.

guitoulefoux committed:
Merge pull request #1 from guitoulefoux/guitoulefoux-patch-1

mysql invalid default value for 'expiry_date'

guitoulefoux committed:
migration from r754 to r1295 TABLE_USERS 

I don't know for what reason but when migrating from r754 to r1295 I noticed that the table "Users "did not have the modifications "Allow Null"

guitoulefoux committed:
migration from r754 to r1295 TABLE_FILES_RELATIONS

I don't know why but when migrating from r754 to r1295 I noticed that the table "TABLE_FILES_RELATIONS" did not have the default value

guitoulefoux committed:
Old files have not name when i download

Migration from r754 to r1295.
When the column "original_url" is null the name file is replaced by "File" when i download it.

ignacionelson commented:
Thanks for your commit!
Nice one on the SQL, but why did you have to modify the file Dopwnload.php? What issue were you getting there?

guitoulefoux commented:
> Thanks for your commit! Nice one on the SQL, but why did you have to modify the file Dopwnload.php? What issue were you getting there?

I don't know by, but while migrating I noticed that when downloading a file, I did not have the name of the file but the name "File" instead of the real name of the file. (probably because my version of projectsend was very old).

Looking at the code I saw that the problem came from the `original_url` column from `tbl_files` which was `NULL`.

![image](https://user-images.githubusercontent.com/51740847/135973702-ded09467-15e2-4a7c-88aa-afa6509c450c.png)

![image](https://user-images.githubusercontent.com/51740847/135973749-6e470df3-0ffb-4eca-9fe8-2f0a6cb3a50c.png)

guitoulefoux committed:
Update core.update.php

ignacionelson committed:
Update Download.php

filename_unfiltered is the raw value on the database, so it's safer to use than filename_original which is filtered by html_output

ignacionelson commented:
Thank you!
